legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/relationalMappingExecution.pure

```diff
@@ -119,6 +119,20 @@ function meta::relational::mapping::generateInstantiationExecutionNode(sq:meta::
    ]);
 }
 
+function meta::relational::mapping::assertCompatibleTypesInRelationalPropertyMapping(r:RelationalPropertyMapping[0..1]):Boolean[1]
+{
+  let classPropertyPureType = $r.property.genericType.rawType;
+  let ropType = meta::relational::mapping::getRelationalTypeFromRelationalPropertyMapping($r, '');
+  let ropCompatiblePureTypes = if($ropType->isEmpty(),
+      | [] 
+     ,| dataTypeToCompatiblePureType($ropType->toOne())->meta::pure::functions::meta::generalizations();
+     );
+
+  assert($ropCompatiblePureTypes->isEmpty() || $classPropertyPureType->isEmpty() ||
+  $ropCompatiblePureTypes->filter(f| $f != Any)->map(m|$classPropertyPureType->toOne()->subTypeOf($m))->contains(true),
+  'Types of class property and its mapped relational property are incompatible');
```

Comment:
Types of class property (<mention type>) and its mapped relational property (<mention type>) are incompatible